Tests/SwiftLintFrameworkTests/ConfigurationTests+MultipleConfigs.swift

```diff
@@ -301,6 +301,122 @@ extension ConfigurationTests {
+            TestCase(optedInInParent: true, disabledInParent: false, optedInInChild: false, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: false, disabledInParent: true, optedInInChild: false, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: true, disabledInParent: true, optedInInChild: false, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: false, disabledInParent: false, optedInInChild: true, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: true, disabledInParent: false, optedInInChild: true, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: false, disabledInParent: true, optedInInChild: true, disabledInChild: true, isEnabled: false),
+            TestCase(optedInInParent: true, disabledInParent: true, optedInInChild: true, disabledInChild: true, isEnabled: false)
+            // swiftlint:enable line_length
+        ]
+        XCTAssertEqual(testCases.unique.count, 4 * 4)
+        let ruleType = ImplicitReturnRule.self
+        XCTAssertTrue(ruleType.isOptInRule)
+        let ruleIdentifier = ruleType.description.identifier
+        for testCase in testCases {
+            let parentConfiguration = Configuration(rulesMode: .default(
+                disabled: testCase.disabledInParent ? [ruleIdentifier] : [],
+                optIn: testCase.optedInInParent ? [ruleIdentifier] : []
+            ))
+            let childConfiguration = Configuration(rulesMode: .default(
+                disabled: testCase.disabledInChild ? [ruleIdentifier] : [],
+                optIn: testCase.optedInInChild ? [ruleIdentifier] : []
+            ))
+            let mergedConfiguration = parentConfiguration.merged(withChild: childConfiguration, rootDirectory: "")
+            let isEnabled = mergedConfiguration.contains(rule: ruleType)
+            XCTAssertEqual(isEnabled, testCase.isEnabled, testCase.message)
+        }
+    }
+
+    func testParentChildDisableForDefaultRule() {
+        struct TestCase: Equatable {
+            let disabledInParent: Bool
+            let disabledInChild: Bool
+            let isEnabled: Bool
+            var message: String {
+                "disabledInParent = \(disabledInParent) disabledInChild = \(disabledInChild)"
+            }
+        }
+        let testCases: [TestCase] = [
+            TestCase(disabledInParent: false, disabledInChild: false, isEnabled: true),
+            TestCase(disabledInParent: true, disabledInChild: false, isEnabled: false),
+            TestCase(disabledInParent: false, disabledInChild: true, isEnabled: false),
+            TestCase(disabledInParent: true, disabledInChild: true, isEnabled: false)
+        ]
+        XCTAssertEqual(testCases.unique.count, 2 * 2)
+        let ruleType = BlanketDisableCommandRule.self
+        XCTAssertFalse(ruleType is OptInRule.Type)
```

Comment:
Either use `ruleType.isOptInRule` here or inline it also in the other two test methods (which I'd prefer over the extension).

Reply:
The problem I was having was that the obvious

```
        let ruleType = ImplicitReturnRule.self
        XCTAssertTrue(ruleType is OptInRule.Type)
```

produces a compilation warning - `'is' test is always true`

I went with

```
        let ruleType = ImplicitReturnRule.self
        XCTAssertTrue((ruleType as Any) is OptInRule.Type)
```

as workaround (unless you have a better one).

Reply:
I see. Fair enough to remove the assertion again. I leave the decision up to you. 😉

Reply:
I think the workaround is not too ugly, so I'm happy to leave it in. I definitely like the idea of validating that the rules are still optin (or not).